script.module.libmediathek3/lib/libmediathek3listing.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+import urllib
+import urllib2
+import socket
+import sys
+import xbmc
+import xbmcplugin
+import xbmcgui
+import xbmcaddon
+import xbmcvfs
+import re
+from datetime import datetime,timedelta
+import time
+
+from libmediathek3utils import clearString
+from libmediathek3utils import getTranslation as translation
+
+icon = xbmc.translatePath(xbmcaddon.Addon().getAddonInfo('icon'))
+fanart = xbmc.translatePath(xbmcaddon.Addon().getAddonInfo('fanart'))
+
+def sortAZ():
+	xbmcplugin.addSortMethod(int(sys.argv[1]), xbmcplugin.SORT_METHOD_TITLE_IGNORE_THE)
+
+def addEntries(l):
+	lists = []
+	doneList = []
+	
+	for d in l:
+		if str(d) in doneList:#primitive methode to filter duplicated entries
+			continue
+		doneList.append(str(d))
+		
+		u = _buildUri(d)
+		newd = {}
+		for key in d:
+			if key.startswith('_'):
+				if isinstance(d[key], unicode):
+					d[key] = d[key].encode('utf-8', 'ignore')
+				newd[key[1:]] = d[key]
+			elif isinstance(d[key], unicode):
+				newd[key] = d[key].encode('utf-8', 'ignore')
+			else:
+				newd[key] = d[key]
+		d = newd
+		
+		if 'type' in d and d['type'] == 'nextPage':
+			d['name'] = translation(31040)
+			if not 'mode' in d:
+				d['mode'] = get_params()['mode']
+		if isinstance(d["name"], unicode):
+			d["name"] = d["name"].encode('utf-8')
+		d["name"] = clearString(d["name"])
+		if 'airedISO8601' in d or 'airedISO8601A' in d:
+			d['aired'],d['airedtime'] = _airedISO8601(d)
+			
+		if 'type' in d and d['type'] == 'date' and 'airedtime' in d:
+			d["name"] = '(' + str(d["airedtime"]) + ') ' + d["name"]
+		elif 'type' in d and d['type'] == 'date' and 'time' in d:
+			d["name"] = '(' + str(d["date"]) + ') ' + d["name"]
+			
+		ilabels = {
+			"Title": clearString(d.get('name','')),
```

Comment:
mainly for skins, and other add-ons might depend on it too.
just fyi, perhaps worth reading: http://forum.kodi.tv/showthread.php?tid=299107